pkg/network/server.go

```diff
@@ -1357,53 +1358,48 @@ func (s *Server) iteratePeersWithSendMsg(msg *Message, send func(Peer, bool, []b
 	if peerN == 0 {
 		return
 	}
-	mrand.Shuffle(peerN, func(i, j int) {
-		peers[i], peers[j] = peers[j], peers[i]
-	})
 	pkt, err := msg.Bytes()
 	if err != nil {
 		return
 	}
 
-	// If true, this node isn't counted any more, either it's dead or we
-	// have already sent an Inv to it.
-	finished := make([]bool, peerN)
-
-	// Try non-blocking sends first and only block if have to.
-	for _, blocking := range []bool{false, true} {
-		for i, peer := range peers {
-			// Send to 2/3 of good peers.
-			if 3*sentN >= 2*(peerN-deadN) {
-				return
-			}
-			if finished[i] {
-				continue
-			}
-			err := send(peer, blocking, pkt)
-			if err == nil {
-				if msg.Command == CMDGetAddr {
-					peer.AddGetAddrSent()
-				}
-				sentN++
-			} else if !blocking && errors.Is(err, errBusy) {
-				// Can be retried.
-				continue
-			} else {
-				deadN++
+	var replies = make(chan error, peerN) // Cache is there just to make goroutines exit faster.
+	var ctx, cancel = context.WithTimeout(context.Background(), s.TimePerBlock/2)
```

Comment:
So currently we have `TimePerBlock/2` to broadcast message to all peers concurrently. Then maybe we need to adjust connection write deadline for TCPPeer wrt this new setting? Currently it's set to `SecondsPerBlock`: https://github.com/nspcc-dev/neo-go/blob/0294e2eb18fedf61b045108eb7f3e63757e855ab/pkg/network/tcp_peer.go#L251

Reply:
I wouldn't do this right in this PR because it requires additional testing, even though it makes some sense.

Reply:
Just a question, [here](https://github.com/nspcc-dev/neo-go/pull/2738#discussion_r991842685) you said that
```
Not an optimization, I'd rather say it's a security fix.
```
but we had this write deadline set, isn't it solves the same problem, just at another level?